tracer/test/Datadog.Trace.Tests/Agent/StatsAggregatorTests.cs

```diff
@@ -99,60 +100,257 @@ public async Task EmptyBuckets()
+
+                aggregator.Add(parentSpan, childSpan, measuredChildSpan1, measuredChildSpan2);
+
+                var buffer = aggregator.CurrentBuffer;
+
+                buffer.Buckets.Should().HaveCount(3);
+
+                foreach (var bucket in buffer.Buckets.Values)
+                {
+                    var topLevelHits = bucket.Key.OperationName == "web.request" ? 1 : 0;
+
+                    bucket.Duration.Should().Be(expectedTotalDuration);
+                    bucket.Hits.Should().Be(1);
+                    bucket.Errors.Should().Be(0);
+                    bucket.TopLevelHits.Should().Be(topLevelHits);
+                    bucket.ErrorSummary.GetCount().Should().Be(0);
+                    bucket.ErrorSummary.GetSum().Should().Be(0);
+                    bucket.OkSummary.GetCount().Should().Be(1.0);
+                    bucket.OkSummary.GetSum().Should().BeApproximately(
+                        expectedOkDuration, expectedOkDuration * bucket.OkSummary.IndexMapping.RelativeAccuracy);
+                }
+            }
+            finally
+            {
+                await aggregator.DisposeAsync();
+            }
+        }
+
+        [Fact]
+        public async Task CollectsTopLevelSpans_TS005()
+        {
+            const int millisecondsToNanoseconds = 1_000_000;
+
+            // All spans should be recorded except snapshotSpan
+            const long expectedTotalDuration = (100 + 200) * millisecondsToNanoseconds;
+            const long expectedOkDuration = (100 + 200) * millisecondsToNanoseconds;
+
+            const long expectedHttpClientTotalDuration = 400 * millisecondsToNanoseconds;
+            const long expectedHttpClientOkDuration = 400 * millisecondsToNanoseconds;
+
+            var start = DateTimeOffset.UtcNow;
+
+            var aggregator = new StatsAggregator(Mock.Of<IApi>(), GetSettings(), Timeout.InfiniteTimeSpan);
+
+            try
+            {
+                var simpleSpan = new Span(new SpanContext(1, 1, serviceName: "service"), start);
+                simpleSpan.SetDuration(TimeSpan.FromMilliseconds(100));
 
-                var measuredChildSpan = new Span(new SpanContext(parentSpan.Context, new TraceContext(Mock.Of<IDatadogTracer>()), "service"), start);
-                measuredChildSpan.SetTag(Tags.Measured, "1");
-                measuredChildSpan.SetDuration(TimeSpan.FromMilliseconds(500));
+                var parentSpan = new Span(new SpanContext(2, 2, serviceName: "service"), start);
+                parentSpan.SetDuration(TimeSpan.FromMilliseconds(200));
 
                 // snapshotSpan shouldn't be recorded, because it has the PartialSnapshot metric (even though it is top-level)
-                var snapshotSpan = new Span(new SpanContext(4, 4, serviceName: "service"), start);
+                var snapshotSpan = new Span(new SpanContext(5, 5, serviceName: "service"), start);
                 snapshotSpan.SetMetric(Tags.PartialSnapshot, 1.0);
```

Comment:
I missed that in the previous implementation. Shall we set this `PartialSnapshot` tag in some cases?

Reply:
Yeah that would be good, what other cases are you thinking of?

Reply:
What I meant is that we have this tag that I didn't know about. We never set it in our code and we test for his presence here (which I do not say we shouldn't in case we add this tag in the feature). So my question was more when should we set this tag (and that question isn't related to this PR I believe)

Reply:
Ah! This is a forward-looking change. Snapshot spans already have an RFC, which is a way of keeping long-running traces alive. Since the backend (currently) terminates a trace after 15 minutes of activity, this feature would send a snapshot/dummy to keep the trace open, and it is identified by having this tag.